### DISCUSSION TITLE
Avocado providing more information via custom assertions

### DISCUSSION THREAD

**Discussion by pevogam** [avocado-framework/avocado#5868] [RFC]
Hi @richtja, @clebergnu, I want to request for some comments regarding the following:

I have isolation tests based on python's `unittest` that I have converted to Avocado tests long ago and run via the Avocado Framework. I recently had a situation where an isolation test took more than an hour to properly debug and could have been significantly faster to debug if the `self.assertEqual` I used from `unittest` would have printed more values and variables involved in the assertion. I then investigated that `pytest` offers more verbosity regarding variables in case of failed assertions but `unittest` is fairly minimal in that regard. I am aware that Avocado dumps some local variables at the end of the test but I could not see all variables I could have inspected right away there. So in that regard I want to ask what do you think about this comparison? Is there any plan or idea to extend Avocado's assertion methods with more information on the variables involved in each one of them similarly to `pytest`? Otherwise one would have to wrap the Avocado `Test` class with another just to achieve this and at the very least this could be something offered for all cases of use.

**Comment by clebergnu** [comment]
Hi @pevogam,

I think adding extra features to the `assert*` methods in `avocado.Test` is a great idea, specially because it doesn't look like it will require API changes.

Can you provide more details on the variables we should be logging?

Thanks!

**Reply by pevogam** [reply]
Something that will be more universal right away would be having all variables in the current scope printed (with some abbreviations on the `repr` is needed). Is this already done fully? At least I see some variables listed in the end but it wasn't all of them.
